server/handler/check_run.go

```diff
@@ -73,7 +73,11 @@ func (h *CheckRun) Handle(ctx context.Context, eventType, deliveryID string, pay
 		pullCtx := pull.NewGithubContext(client, fullPR)
 
 		logger := logger.With().Int(githubapp.LogKeyPRNum, pr.GetNumber()).Logger()
-		if err := h.ProcessPullRequest(logger.WithContext(ctx), pullCtx, client, fullPR); err != nil {
+		config, err := h.FetchConfig(ctx, client, pr)
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch configuration")
```

Comment:
Because `FetchConfig` already wraps errors with a similar message, wrapping them again doesn't add useful context.
```suggestion
			return err
```